src/DurableTask.Netherite/Connections/ConnectionInfo.cs

```diff
@@ -0,0 +1,152 @@
+        /// </summary>
+        /// <param name="connectionString">The connection string.</param>
+        /// <returns>The connection info.</returns>
+        public static ConnectionInfo FromEventHubsConnectionString(string connectionString)
+        {
+            var properties = Azure.Messaging.EventHubs.EventHubsConnectionStringProperties.Parse(connectionString);
+            string hostName = properties.FullyQualifiedNamespace;
+            string nameSpaceName = hostName.Split('.')[0];
+
+            return new ConnectionInfo()
+            {
+                ResourceName = nameSpaceName,
+                ConnectionString = connectionString,
+                TokenCredential = null,
+                HostName = hostName,
+                Scopes = s_eventhubs_scopes,
+            };
+        }
+
+        /// <summary>
+        /// Creates a connection info from a storage connection string
+        /// </summary>
+        /// <param name="connectionString">The connection string.</param>
+        /// <param name="resourceType">The resource type being accessed.</param>
+        /// <returns>The connection info.</returns>
+        public static ConnectionInfo FromStorageConnectionString(string connectionString, ConnectionResolver.ResourceType resourceType)
+        {
+            var cloudStorageAccount = CloudStorageAccount.Parse(connectionString);
+            return new ConnectionInfo()
+            {
+                ResourceName = cloudStorageAccount.Credentials.AccountName,
+                ConnectionString = connectionString,
+                TokenCredential = null,
+                HostName = GetEndpoint().Host,
+                Scopes = s_storage_scopes,
+            };
+
+            Uri GetEndpoint() => resourceType == ConnectionResolver.ResourceType.TableStorage 
+                ? cloudStorageAccount.TableEndpoint : cloudStorageAccount.BlobEndpoint;
+        }
+
+        /// <summary>
+        /// Creates a connection info from a token credential.
+        /// </summary>
+        /// <param name="tokenCredential">The token credential.</param>
+        /// <param name="name">The name of the resource (account name or namespace name).</param>
+        /// <param name="resourceType">The type of the resource.</param>
+        /// <returns></returns>
+        /// <returns>The connection info.</returns>
+        public static ConnectionInfo FromTokenCredential(Azure.Core.TokenCredential tokenCredential, string name, ConnectionResolver.ResourceType resourceType)
+        {
+            switch (resourceType)
+            {
+                case ConnectionResolver.ResourceType.EventHubsNamespace:
+                    {
+                        return new ConnectionInfo()
+                        {
+                            ResourceName = name,
+                            ConnectionString = null,
+                            TokenCredential = tokenCredential,
+                            HostName = $"{name}.servicebus.windows.net",
```

Comment:
if we want to support Netherite in other Azure clouds we will need some follow up work for cloud-specific host names. Or can this be adjusted to take in the full hostname instead of name?

Reply:
Yes, the hardcoded string here is only meant to be a convenient default.
The connection resolver can set this any other hostname.